elastalert/ruletypes.py

```diff
@@ -535,17 +535,18 @@ def get_all_terms(self, args):
         else:
             end = ts_now()
         start = end - window_size
-        if self.rules.get('use_strftime_index'):
-            index = format_index(self.rules['index'], start, end)
-        else:
-            index = self.rules['index']
-        time_filter = {self.rules['timestamp_field']: {'lte': dt_to_ts(end), 'gte': dt_to_ts(start)}}
-        query_template['filter'] = {'bool': {'must': [{'range': time_filter}]}}
-        query = {'aggs': {'filtered': query_template}}
+        step = datetime.timedelta(**self.rules.get('window_step_size', {'days': 1}))
 
         for field in self.fields:
+            tmp_start = start
+            tmp_end = min(start + step, end)
+
+            time_filter = {self.rules['timestamp_field']: {'lt': dt_to_ts(tmp_end), 'gte': dt_to_ts(tmp_start)}}
```

Comment:
I think people expect inclusive on both ends more often than not, though there is no real consistency throughout elastalert unfortunately.